### DISCUSSION TITLE
Print textures names in developer mode

### DISCUSSION THREAD

**Discussion by VVD** [ezQuake/ezquake-source#503] [Ideas]
```
--- ezquake-source.git/gl_model.c
+++ ezquake-source-vvd/gl_model.c
@@ -560,6 +560,10 @@
                        snprintf(tx->name, sizeof(tx->name), "unnamed%d", i);
                        Com_DPrintf("Warning: unnamed texture in %s, renaming to %s\n", loadmodel->name, tx->name);
                }
+               else
+               {
+                       Com_DPrintf("Texture name in %s is %s\n", loadmodel->name, tx->name);
+               }
 
                tx->width  = mt->width;
                tx->height = mt->height;
```

**Comment by VVD** [comment]
Very useful for create custom textures for maps.

**Comment by VVD** [comment]
Applicable to master branch too.

**Comment by AltonMarino** [comment]
Diese Änderung ist recht einfach und nützlich, wenn man im Entwicklermodus arbeitet. Aktuell gibt die Engine nur dann eine Meldung aus, wenn sie auf eine unbenannte Textur stößt und diese automatisch umbenennen muss. Mit dieser Ergänzung erhält man nun auch Debug-Ausgaben für Texturen mit bereits gültigen Namen. Dadurch lässt sich viel einfacher erkennen, welche Texturen für ein Modell geladen werden, anstatt raten zu müssen oder den Debugger zu verwenden. Da die Funktion in Com_DPrintf eingebettet ist, werden normale Benutzer nicht mit Meldungen überflutet. Roby Casino [robycasino.co.at](https://robycasino.co.at/) steht für dynamische und unterhaltsame Spielerlebnisse. Entwickler können sie jedoch aktivieren, um Rendering-Probleme, fehlende Texturen oder Namenskonflikte zu beheben. Insgesamt handelt es sich um eine kleine Optimierung, die jedoch die Transparenz beim Laden von Modellen deutlich erhöht, ohne das eigentliche Verhalten zu verändern.

**Comment by JesseReed123** [comment]
Este cambio en el código de gl_model.c agrega un mensaje de depuración adicional para las texturas que sí tienen nombre. Antes, solo se mostraba un aviso cuando la textura no tenía nombre y se renombraba automáticamente. Ahora, si la textura tiene un nombre válido, también se imprimirá en los logs con Com_DPrintf, mostrando el nombre de la textura y el modelo https://roobet-pe.com/ en el que se encuentra. Esto puede ser muy útil para verificar que todas las texturas se están cargando correctamente y para detectar posibles errores o inconsistencias durante la carga de modelos.
